### DISCUSSION TITLE
Is there a practical limit on the number of tracks per album?

### DISCUSSION THREAD

**Discussion by troycarpenter** [ampache/ampache#3624] [Q&A]
I have a box set of around 100 CDs with about 3000 total songs across all discs.  When I display this album in the GUI, it is almost painfully slow before the content shows.  Dsub (a *sonic client) sometimes locks up while trying to enumerate all the tracks even if split into multiple discs.

To mitigate that, I've had to create multiple albums, listed as volumes.  Unfortunately, the album listings become full of these extra albums instead of having just one entry.

Since multiple discs are still combined as a single track list (no matter how it's presented to the user), is there some practical limit as to how many tracks should be in an album?

**Comment by lachlan-00** [comment]
it gets the objects and does a browse on each disk in the album. it really shouldn't be too slow to do 500 disks. i'll see if i can make a bigger one

the largest album i have is 414 songs over 33 cd's
It takes about 2 seconds to load in the web interface and in dsub

**Comment by troycarpenter** [comment]
It seems to me the lag is due to the number of tracks, not necessarily the number of discs.  And Dsub on the mobile devices sometimes times out trying to get all that data.

**Reply by lachlan-00** [reply]
could definitely be combo of things but it's not obvious.

I just put 740 songs into a single album by editing the DB and it loaded just as quick as my 400 track classical album.

I'm not seeing lag around that size but as a goof i put 480k songs into a single album and then refreshed.....

it's still loading but took about 3 minutes to get to the second cd

One thing that might help me find what it's doing is all the queries in the Query class

replace `//debug_event` with `//debug_event` in `src/Repository/Model/Query.php`

It might give me an idea of which parts are the slow areas in your browse
